python/sglang/srt/layers/quantization/fp8.py

```
@@ -347,7 +348,25 @@ def process_weights_after_loading(self, layer: Module) -> None:
                 )
                 return
             else:
+                # For fp8 linear weights run with deepgemm, the weights and scales need be requantized to ue8m0
+                from sglang.srt.model_loader.utils import (
+                    should_deepgemm_weight_requant_ue8m0,
+                )
+
+                if should_deepgemm_weight_requant_ue8m0(
+                    weight_block_size=getattr(
+                        self.quant_config, "weight_block_size", None
+                    ),
+                    layer=layer,
+                ) and not getattr(layer, "_executed_weight_requant_ue8m0", False):
```

nit: I personally prefer setting `layer._executed_weight_requant_ue8m0=False` at initialization stage, s.t. we do not need `getattr`. b/c if we use getattr and the field is somehow renamed (or we misspell the field name), it will silently fail, which is a bit harder to debug

---

Changed to another way avoiding getattr